src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -129,6 +139,12 @@ public async Task UpdateSearchParameterAsync(ITypedElement searchParam, RawResou
                     throw new SearchParameterNotSupportedException(searchParameterInfo.Url);
                 }
 
+                // check data store specific support for SearchParameter
+                if (!_dataStoreSearchParameterValidator.ValidateSearchParameter(searchParameterInfo, out var errorMessage))
```

Comment:
More of a question, but does SQL support all of these? https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs#L298